src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py

```diff
@@ -6,64 +6,196 @@
+            ##########|  logo  |##########      |
+            ##########|________|##########      |-------> x (pitch)
+            ##############################     /
+            ##############################    /
+            ##############################   /
+            ##############################  z (roll)
+
+        - blur: A boolean flag indicating whether augmented images should be
+                blurred or not. True by default.
+        - noise: A boolean flag indicating whether augmented images should have
+                artificial noise or not. True by default.
+        - lighting_perturbation: A boolean flag indicating whether the 
+                augmented images should have their lightings perturbed or not.
+                True by default.
+    """
+    if not isinstance(target, str):
+        raise TypeError("'target' must be of type string")
+    if isinstance(dataset, _tc.SFrame):
+        image_column_name = _tkutl._find_only_image_column(dataset)
+        target_column_name = target
+        dataset_to_augment = dataset
+    elif isinstance(dataset, _tc.Image):
+        image_column_name = "image"
+        target_column_name = "target"
+        dataset_to_augment = _tc.SFrame({image_column_name: dataset,
+                                         target_column_name: target})
+    else:
+        raise TypeError("'dataset' must be of type SFrame or Image")
+
+    one_shot_model = _extensions.one_shot_object_detector()
     if seed is None: seed = _random.randint(0, 2**32 - 1)
     if backgrounds is None:
         # replace this with loading backgrounds from developer.apple.com
         backgrounds = _tc.SArray()
     # Option arguments to pass in to C++ Object Detector, if we use it:
     # {'mlmodel_path':'darknet.mlmodel', 'max_iterations' : 25}
-    augmented_data = model.augment(dataset, target, backgrounds, {"seed":seed})
-    od_model = _tc.object_detector.create(augmented_data)
-    state = {'detector':od_model}
-    return OneShotObjectDetector(state)
+    options_for_augmentation = {
+        "seed": seed,
+        "flip_horizontal": flip_horizontal,
+        "flip_vertical": flip_vertical,
+        "kwargs": kwargs
+    }
+    augmented_data = one_shot_model.augment(dataset_to_augment,
+                                            image_column_name,
+                                            target_column_name,
+                                            backgrounds,
+                                            options_for_augmentation)
+    model = _tc.object_detector.create( augmented_data,
+                                        batch_size=batch_size,
+                                        max_iterations=max_iterations,
+                                        verbose=verbose)
+    return OneShotObjectDetector(model.__proxy__)
 
 
-class OneShotObjectDetector(_CustomModel):
-    _PYTHON_ONE_SHOT_OBJECT_DETECTOR_VERSION = 1
-    
+class OneShotObjectDetector(_ObjectDetector):
```

Comment:
I thought we decided that the new class should _not_ inherit from the existing `ObjectDetector` type. We don't want to blindly forward the entire API surface.

Reply:
Sorry, I was under the impression that the new class should inherit from the existing ObjectDetector, so that the model we get behaves like an Object Detector. How would we ensure that the model we get out of `.create` is an `ObjectDetector` and not a `OneShotObjectDetector`?